mde-services/src/main/java/de/terrestris/mde/mde_backend/controller/MetadataCollectionController.java

```diff
@@ -605,25 +608,43 @@ public ResponseEntity<Void> deleteComment(
             description =
                 "Not found: The provided metadata collection does not exist (or you don't have the permission to delete it)",
             content = @Content),
+        @ApiResponse(
+            responseCode = "409",
+            description =
+                "Conflict: The metadata collection is assigned to a user or you are not the owner or in the team and cannot be deleted",
+            content = @Content),
         @ApiResponse(
             responseCode = "500",
             description =
                 "Internal Server Error: Something internal went wrong while deleting the metadata collection",
             content = @Content)
       })
-  public ResponseEntity<?> delete(@PathVariable("metadataId") String metadataId) {
+  public ResponseEntity<?> delete(
+      @PathVariable("metadataId") String metadataId, Authentication auth) {
     try {
-      Optional<MetadataCollection> metadataCollection = service.findOneByMetadataId(metadataId);
-
-      if (metadataCollection.isEmpty()) {
-        log.warn("Could not find metadata collection with ID {}", metadataId);
-        return new ResponseEntity<>(NOT_FOUND);
+      MetadataCollection metadataCollection =
+          service
+              .findOneByMetadataId(metadataId)
+              .orElseThrow(() -> new ResponseStatusException(NOT_FOUND));
+      if (metadataCollection.getAssignedUserId() != null) {
+        return new ResponseEntity<>(CONFLICT);
+      }
+      if (!auth.getAuthorities().contains(new SimpleGrantedAuthority("MdeAdministrator"))) {
```

Comment:
This check might fail since [!75](https://github.com/gdi-be/mde-backend/pull/75) is merged right now. The expected role is formatted like `ROLE_MDEADMINISTRATOR`.

Reply:
Right, nice find, fixed